### PR TITLE
ICU-21404 Migrate some process docs from icusite to Markdown

### PR DESCRIPTION
**Pull request by hugovdm**
This PR branch was being served at: https://hugovdm.github.io/icu/
(Latest squashing reverts an edit needed to make the links work on hugovdm.*)

Markdown page / origin on sites:
- [announce.md](https://hugovdm.github.io/icu/processes/release/tasks/announce.html) / https://sites.google.com/site/icusite/processes/release/tasks/announce
- [build.md](https://hugovdm.github.io/icu/processes/release/tasks/build.html) / https://sites.google.com/site/icusite/processes/release/tasks/build
- [copy-shared-tests.md](https://hugovdm.github.io/icu/processes/release/tasks/copy-shared-tests.html) / https://sites.google.com/site/icusite/processes/release/tasks/copy-shared-tests
- [docs.md](https://hugovdm.github.io/icu/processes/release/tasks/docs.html) / https://sites.google.com/site/icusite/processes/release/tasks/docs
- [healthy-code.md](https://hugovdm.github.io/icu/processes/release/tasks/healthy-code.html) / https://sites.google.com/site/icusite/processes/release/tasks/healthy-code
- [index.md](https://hugovdm.github.io/icu/processes/release/tasks/) / https://sites.google.com/site/icusite/processes/release/tasks
- [integration.md](https://hugovdm.github.io/icu/processes/release/tasks/integration.html) / https://sites.google.com/site/icusite/processes/release/tasks/integration
- [legal/index.md](https://hugovdm.github.io/icu/processes/release/tasks/legal/) / https://sites.google.com/site/icusite/processes/release/tasks/legal
- [miscellaneous.md](https://hugovdm.github.io/icu/processes/release/tasks/miscellaneous.html) / https://sites.google.com/site/icusite/processes/release/tasks/miscellaneous
- [publish/index.md](https://hugovdm.github.io/icu/processes/release/tasks/publish/) / https://sites.google.com/site/icusite/processes/release/tasks/publish
- [release-build.md](https://hugovdm.github.io/icu/processes/release/tasks/release-build.html) / https://sites.google.com/site/icusite/processes/release/tasks/release-build
- [standards.md](https://hugovdm.github.io/icu/processes/release/tasks/standards.html) / https://sites.google.com/site/icusite/processes/release/tasks/standards
- [versions.md](https://hugovdm.github.io/icu/processes/release/tasks/versions.html) / https://sites.google.com/site/icusite/processes/release/tasks/versions
- [unicode-update.md](https://hugovdm.github.io/icu/processes/unicode-update.html) / https://sites.google.com/site/icusite/processes/uni-up

##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21404
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [ ] Tests included
- [X] Documentation is changed or added

### CODE REVIEW

File: docs/processes/release/tasks/standards.md
Comment (markusicu) [inline docs/processes/release/tasks/standards.md:14]: svn links to .../repos/icu/data/... are in a separate repo: https://github.com/unicode-org/icu-data

This one is https://github.com/unicode-org/icu-data/blob/master/tzdata/readme.txt

File: docs/processes/uni-up.md
Comment (markusicu) [inline docs/processes/uni-up.md:25]: https://github.com/unicode-org/icu/blob/master/icu4c/source/data/unidata/changes.txt

File: docs/processes/uni-up.md
Comment (markusicu) [inline docs/processes/uni-up.md:43]: https://github.com/unicode-org/icu/tree/master/tools/unicode

still somewhat separate from ICU4C/ICU4J, but no longer in a separate repo

File: docs/processes/uni-up.md
Comment (markusicu) [inline docs/processes/uni-up.md:39]: yet another repo...  https://github.com/unicode-org/icu-docs/tree/master/design/properties/genudata

File: docs/processes/release/tasks/standards.md
Comment (markusicu) [inline docs/processes/release/tasks/standards.md:34]: should the `@` be escaped? it shows up in red.

File: docs/processes/release/tasks/standards.md
Comment (markusicu) [inline docs/processes/release/tasks/standards.md:27]:
```suggestion
[icu4c/source/data/unidata/changes.txt](https://github.com/unicode-org/icu/blob/master/icu4c/source/data/unidata/changes.txt)
```

File: docs/processes/release/tasks/standards.md
Comment (markusicu) [inline docs/processes/release/tasks/standards.md:47]:
```suggestion
[icu4c/source/data/unidata/changes.txt](https://github.com/unicode-org/icu/blob/master/icu4c/source/data/unidata/changes.txt)
```

File: docs/processes/release/tasks/standards.md
Comment (markusicu) [inline docs/processes/release/tasks/standards.md:63]: remove **bold** markers

**Comment by markusicu** [comment]
@echeran FYI  some boring docs about Unicode updates that I mostly forgot about...

File: docs/processes/uni-up.md
Comment (hugovdm) [reply docs/processes/uni-up.md]: Done.

File: docs/processes/uni-up.md
Comment (hugovdm) [reply docs/processes/uni-up.md]: Done.

File: docs/processes/release/tasks/standards.md
Comment (hugovdm) [reply docs/processes/release/tasks/standards.md]: Done.

File: docs/processes/release/tasks/standards.md
Comment (hugovdm) [reply docs/processes/release/tasks/standards.md]: Done.

File: docs/processes/release/tasks/standards.md
Comment (hugovdm) [reply docs/processes/release/tasks/standards.md:34]: Looks like it's an "@mention". Backslash escaping apparently doesn't work, backtick apparently does: https://github.com/github/markup/issues/1168

File: docs/processes/uni-up.md
Comment (hugovdm) [reply docs/processes/uni-up.md:6]: Done.

File: docs/processes/release/tasks/standards.md
Comment (hugovdm) [reply docs/processes/release/tasks/standards.md]: Suggestion committed, thanks!

File: docs/processes/release/tasks/standards.md
Comment (hugovdm) [reply docs/processes/release/tasks/standards.md]: Suggestion committed, thanks!

**Comment by hugovdm** [comment]
No edits on the originals:
- https://sites.google.com/site/icusite/processes/release/tasks/standards
- https://sites.google.com/site/icusite/processes/uni-up

Edits are limited enough that I'll accept squashing all into one commit, instead of two commits (auto import then edit). I'll separately keep a reference to what the edits were, for potential convenience of further migration.

**Comment by hugovdm** [comment]
Current Status:
- initial machine-converstion is in this PR, but review and possible style improvements still need to be made,
- also some improvements to sitenav, iirc,
- then checking for any further updates to the Google Site, since I did the conversion.

(Currently overwhelmed with non-ICU tasks, I do intend to deal with this before end of May!)

**Review by sven-oly — COMMENTED** [review]
This is mostly OK for a straight port. Thanks for doing this. Here are some formatting issues

a. I see instances of "{: .no_toc }" that seem wrong, e.g., announce.md

b. In some, there are distinct steps that now render as short paragraphs. I suggest that these should be numbered, or at least made into bulleted lists so they can be distinguished from comments.

c. Are the explicit numbers for steps (build.md) necessary or can markdown authomatically number these? That would make updating much easier.

d. integration.md: In section "Run ICU4J Locale Service Provider tests", the output should be put in a code block

e. In legal/index.md, just remove the obsolete references.

f. In publish/index.md, fix this - it appears as text but is probably meant to describe the image display: {width="320" height="171"}

g. release-build.md: Blocks such as "Create maven local setting file" should appears as code blocks instead of disconnected quoted text. Format issue?

h. versions.md: some of the numbered sections under "ICU4J" have the number on a separate line from the content.  Also, the section "time bomb" should be a block without fill.
 
I suggest a major review of all these. Perhaps we should use additional PRs for these changes.

1. Some references are way out of data and probably should be removed.
2. announce.md: freshmeat.net: "Effective 2014-06-18 Freecode is no longer being updated (content may be stale)"
3. announce.md: http://sourceforge.net/projects/icu - we haven't updated there since version 61. Should we consider removing this reference?
4. When can we remove notes on building older versions?

5. unicode-update.md:
  "Last thorough review was in 2010" !!

File: docs/processes/release/tasks/integration.md
Comment (markusicu) [inline docs/processes/release/tasks/integration.md:39]: code block

currently runs into a big sausage, see https://hugovdm.github.io/icu/processes/release/tasks/integration.html

File: docs/processes/release/tasks/legal/index.md
Comment (markusicu) [inline docs/processes/release/tasks/legal/index.md:51]: Please refer to 
`[svn changeset r39632](https://github.com/unicode-org/icu/commit/0001f6c5e92f6f3a8d66c7dbc47cc24df7633a71) ([ICU-12895](https://unicode-org.atlassian.net/browse/ICU-12895))`

(FYI  Found via `git log --grep=39632`)

File: docs/processes/release/tasks/publish/index.md
Comment (markusicu) [inline docs/processes/release/tasks/publish/index.md:113]: The width and height show up as literal text on the rendered page. Incompatible syntax?

File: docs/processes/release/tasks/release-build.md
Comment (markusicu) [inline docs/processes/release/tasks/release-build.md:125]: code block

File: docs/processes/release/tasks/release-build.md
Comment (markusicu) [inline docs/processes/release/tasks/release-build.md:164]: some of the sample outputs and configs etc. have funky formatting but are readable...

File: docs/processes/release/tasks/release-build.md
Comment (markusicu) [inline docs/processes/release/tasks/release-build.md:261]: This image does not render.

File: docs/processes/release/tasks/versions.md
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:131]: code block

https://hugovdm.github.io/icu/processes/release/tasks/versions.html

File: docs/processes/release/tasks/versions.md
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:178]: code block

File: docs/processes/release/tasks/versions.md
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:185]: make sure that the list works

File: docs/processes/release/tasks/versions.md
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:204]: code block

File: docs/processes/release/tasks/versions.md
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:210]: Please fix links to ICU tickets --> Jira

**Review by markusicu — COMMENTED** [review]
Hi Hugo, sorry again for a late review, and thanks for pulling in more pages here.

Your changes for my previous comments look good.

I glanced through the rendered pages linked from https://hugovdm.github.io/icu/processes/release/tasks/ -- didn't check if that's all of the .md files here.

At a glance, mostly looks good. Some formatting is not pretty but readable; let's not look for perfection here. I made specific comments for what looks more broken.

Is this the complete set of release task pages? It would be good to finish this up soon, start pointing from the Sites pages to these here, and tackle the next round of Sites pages in another PR.

**Comment by markusicu** [comment]
> This is mostly OK for a straight port. Thanks for doing this. Here are some formatting issues

Let's get the contents over, and make it readable and usable -- but let's not substantially change contents here and now (e.g., deleting struck-out text and other, even obsolete, text) and let's not fine-tune formatting beyond readable/usable.

Better to get something in and be able to switch over than not finish due to imperfections.

Yes, we should review the contents, but let's do that in more focused follow-up PRs. Some of it can be part of doing BRS tasks.

**Comment by hugovdm** [comment]
Thank you for the reviews - I still had this marked as draft and "waiting on author" because I had not done good review myself yet. (Last time I stopped working on this, I was looking at how to test locally, so I can iterate on the sidebar links/navigation more quickly. Now tackling content review first, tracking progress:

- [ ] [announce.md](https://hugovdm.github.io/icu/processes/release/tasks/announce.html) / https://sites.google.com/site/icusite/processes/release/tasks/announce
- [ ] [build.md](https://hugovdm.github.io/icu/processes/release/tasks/build.html) / https://sites.google.com/site/icusite/processes/release/tasks/announce
- [ ] [copy-shared-tests.md](https://hugovdm.github.io/icu/processes/release/tasks/copy-shared-tests.html) / https://sites.google.com/site/icusite/processes/release/tasks/announce
- [ ] [docs.md](https://hugovdm.github.io/icu/processes/release/tasks/docs.html) / https://sites.google.com/site/icusite/processes/release/tasks/docs
- [ ] [health-code.md](https://hugovdm.github.io/icu/processes/release/tasks/healthy-code.html) / https://sites.google.com/site/icusite/processes/release/tasks/healthy-code
- [ ] [index.md](https://hugovdm.github.io/icu/processes/release/tasks/) / https://sites.google.com/site/icusite/processes/release/tasks
- [ ] [integration.md](https://hugovdm.github.io/icu/processes/release/tasks/integration.html) / https://sites.google.com/site/icusite/processes/release/tasks/integration
- [ ] [legal/index.md](https://hugovdm.github.io/icu/processes/release/tasks/legal/) / https://sites.google.com/site/icusite/processes/release/tasks/legal
- [ ] [miscellaneous.md](https://hugovdm.github.io/icu/processes/release/tasks/miscellaneous.html) / https://sites.google.com/site/icusite/processes/release/tasks/miscellaneous
- [ ] [publish/index.md](https://hugovdm.github.io/icu/processes/release/tasks/publish/) / https://sites.google.com/site/icusite/processes/release/tasks/publish
- [ ] [release-build.md](https://hugovdm.github.io/icu/processes/release/tasks/release-build.html) / https://sites.google.com/site/icusite/processes/release/tasks/release-build
- [ ] [standards.md](https://hugovdm.github.io/icu/processes/release/tasks/standards.html) / https://sites.google.com/site/icusite/processes/release/tasks/standards
- [ ] [versions.md](https://hugovdm.github.io/icu/processes/release/tasks/versions.html) / https://sites.google.com/site/icusite/processes/release/tasks/versions
- [ ] [unicode-update.md](https://hugovdm.github.io/icu/processes/unicode-update.html) / https://sites.google.com/site/icusite/processes/uni-up

File: docs/processes/release/tasks/versions.md
Comment (hugovdm) [reply docs/processes/release/tasks/versions.md:132]: Done.

File: docs/processes/release/tasks/versions.md
Comment (hugovdm) [reply docs/processes/release/tasks/versions.md:179]: Done - using <pre> in this file, because some blocks need <b> (used <b> on Sites).

File: docs/processes/release/tasks/versions.md
Comment (hugovdm) [reply docs/processes/release/tasks/versions.md:203]: Done.

File: docs/processes/release/tasks/versions.md
Comment (hugovdm) [reply docs/processes/release/tasks/versions.md]: Done.

File: docs/processes/release/tasks/versions.md
Comment (hugovdm) [reply docs/processes/release/tasks/versions.md]: Done.

File: docs/processes/release/tasks/integration.md
Comment (hugovdm) [reply docs/processes/release/tasks/integration.md:43]: Done - desausageified, various other fixes to this page's markdown.

Text with orange background doesn't look as nice here: I can do a <span>, but not a <p>, as the latter breaks markdown.

File: docs/processes/release/tasks/legal/index.md
Comment (hugovdm) [reply docs/processes/release/tasks/legal/index.md]: s/X-SVN-Rev: 39632/svn changeset r39632/ - done. (I caught that when fixing other /trac/ links. Couldn't find all changesets though, left a comment about the changeset I couldn't find.)

File: docs/processes/release/tasks/legal/find_textfiles
Comment (hugovdm) [inline docs/processes/release/tasks/legal/find_textfiles:4]: @yumaoka I suppose I should just delete this file?
- Author is listed below as Umesh Nair.
- Copyright notice years I based on https://sites.google.com/site/icusite/system/app/pages/admin/revisions?wuid=wuid:gx:305064f7a236995f which has two versions of the script, uploaded by you.

File: docs/processes/release/tasks/legal/index.md
Comment (hugovdm) [inline docs/processes/release/tasks/legal/index.md:40]: Script linked here (@yumaoka).

I didn't want to think about what to update as part of the migration PR. And I didn't want to poke into the perl script to see if there's anything interesting in it. :)

https://sites.google.com/site/icusite/processes/release/tasks/legal

**Comment by hugovdm** [comment]
publish/index.md is almost good - I want to give it another look-over. Then two more pages, and reviewing the navigation again / fixing it (iirc it's not quite right yet). I'll try to finish this tomorrow (today's full).

File: docs/processes/release/tasks/publish/index.md
Comment (hugovdm) [reply docs/processes/release/tasks/publish/index.md]: Done. (Deleted.)

File: docs/processes/release/tasks/integration.md
Comment (markusicu) [reply docs/processes/release/tasks/integration.md:43]: Great that you figured out how to do the orange background... but (in the future) if it's too onerous, it should be ok to use simpler means of emphasis, such as italics or bold.

File: docs/processes/release/tasks/release-build.md
Comment (hugovdm) [reply docs/processes/release/tasks/release-build.md]: Done.

File: docs/processes/release/tasks/release-build.md
Comment (hugovdm) [reply docs/processes/release/tasks/release-build.md]: Done.

File: docs/processes/release/tasks/integration.md
Comment (hugovdm) [reply docs/processes/release/tasks/integration.md:43]: Agreed: just didn't want to think too much about it for this conversion, and trusting future editors to evaluate the presentation. :)

File: docs/processes/release/tasks/release-build.md
Comment (hugovdm) [reply docs/processes/release/tasks/release-build.md]: I added the png as NexusRepoMan.png, and I'm not touching setup/java/ant/ yet, so I added it in the release/tasks/ directory.

File: docs/_config.yml
Comment (echeran) [inline docs/_config.yml:13]: I usually forget to remove the url override before merging, so what I started doing is leave this setting alone in the PR branch, and then I maintain a "demo branch" that rebases [a single commit](https://github.com/echeran/icu/commit/1c1b0427177d212fcf4c49504959b2543d0ea2bb) on top of the PR branch. And then I configure GH pages to always point to the demo branch. It comes at the cost of remembering that every push to the PR branch requires a rebase and push of the demo branch, too.

You may find that this process is more effort than it's worth, so do whatever works best for you.

**Review by echeran — COMMENTED** [review]
General observation + question : I noticed that the markdown files in `/docs/processes` don't show up in the navigation bar. But the files do get rendered in the output website ([example](https://unicode-org.github.io/icu/processes/rules_update)).

Do we want to have these processes pages represented in the left-hand side navigation menu? 

If so, the first thing is to add a `./docs/processes/index.md` file, similar to all of the `index.md` in the other subdirectories. I also noticed that more than half of the files in `/docs/processes` have the Kramdown header content specific for our theme. That header info is pretty much for configuring names and ordering of menus and sub-menus. Our theme limits the sub-menus to a depth-of 2 (grandchildren of top-level menu items).

**Comment by hugovdm** [comment]
Thanks! I'll take another look at the nav and make it more consistent!

File: docs/_config.yml
Comment (hugovdm) [reply docs/_config.yml:13]: Thanks for the tip! I'll consider my options and see what works best for me!

**Comment by hugovdm** [comment]
It is fuzzer_targets.md and rules_update.md that have no links. I don't know what links to these pages, it doesn't look like they're liked in BRS. Where in the nav do you think these two should appear? And/or can we leave this as not a part of this PR, since this PR is about adding and linking to new migrated docs?

In this PR, I have placed unicode-update.md under "Release & Milestone Tasks" as well, even though it doesn't seem to appear in the BRS spreadsheet. (It also hasn't been updated since 2010.) I wondered whether to rather link it from its parent ("Contributors")...

The old site's navigvation (http://site.icu-project.org/system/app/pages/sitemap/hierarchy) had:
ICU Processes and Procedures > Throwing the Big Red Switch: How to ship ICU > Release & Milestone Tasks
I've placed "Release & Milestone Tasks" directly under "Contributors" for the time being. Continue as is, to get this merged, or make further adjustments to how we structure this sidebar nav?

**Comment by hugovdm** [comment]
There is information at these two addresses:
- https://sites.google.com/site/icusite/processes/
- https://sites.google.com/site/icusite/processes/release/

which should arguably become:
- https://hugovdm.github.io/icu/processes/index.md
- https://hugovdm.github.io/icu/processes/release/index.md

I propose we leave the left-side nav in this PR as is, and update it when these two documents are* migrated. (*Likely "when I migrate these two docs". ;-) )

**Comment by hugovdm** [comment]
Or let me state this clearly:

> Do we want to have these processes pages represented in the left-hand side navigation menu?

This PR already adds all its new documents to the left-hand side navigation menu. It doesn't add anything else to the nav menu, since my focus was just getting new documents migrated with minimum debate. Can further improvements to the nav menu be done in a separate PR so that I can at least go ahead and deprecate the Sites pages, update links in BRS sheets, etc?

**Comment by markusicu** [comment]
> It is fuzzer_targets.md and rules_update.md that have no links. I don't know what links to these pages, it doesn't look like they're liked in BRS. Where in the nav do you think these two should appear? And/or can we leave this as not a part of this PR, since this PR is about adding and linking to new migrated docs?

Follow-up PR is fine.

> In this PR, I have placed unicode-update.md under "Release & Milestone Tasks" as well, even though it doesn't seem to appear in the BRS spreadsheet. (It also hasn't been updated since 2010.) I wondered whether to rather link it from its parent ("Contributors")...

"Release & Milestone Tasks" is right in principle. Once this is in, I should look at updating the page and cross-linking it with my work log.

> The old site's navigvation (http://site.icu-project.org/system/app/pages/sitemap/hierarchy) had:
> ICU Processes and Procedures > Throwing the Big Red Switch: How to ship ICU > Release & Milestone Tasks
> I've placed "Release & Milestone Tasks" directly under "Contributors" for the time being. Continue as is, to get this merged, or make further adjustments to how we structure this sidebar nav?

Suggestion:
- New top page "Processes & Procedures" with the contents of http://site.icu-project.org/processes
- Put that at the end of the nav bar
- Hoist the contents of http://site.icu-project.org/processes/release into the "ICU Release Process" section of that page
  - Append ": Throwing the Big Red Switch" to the section heading
  - Omit the "Old stuff" under the horizontal line
- Intermediate "Release & Milestone Tasks" page just as a navigation node
  - Empty page like now
  - This keeps us from wanting to prefix something like this page title to the prefix of each BRS page title
  - I am open to other suggestions
- Task pages under that, on second=final nav bar level

**Comment by markusicu** [comment]
> This PR already adds all its new documents to the left-hand side navigation menu. It doesn't add anything else to the nav menu (i.e. doesn't try to fix things currently broken on the ICU github pages), since my focus was just on getting new documents migrated, with minimum debate and minimum changes. Can further improvements to the nav menu be done in a separate PR so that I can at least go ahead and deprecate the Sites pages, update links in BRS sheets, etc?

Yes, that's ok.

**Comment by markusicu** [comment]
Hugo: When you think that these pages are in good-enough shape, I am happy to rubber-stamp this PR so we can move forward.

**Comment by markusicu** [comment]
Oh, and since there are (I believe) still just minimal contents changes, are you planning to squash this PR into one commit?

**Comment by hugovdm** [comment]
Force-pushed twice:
- First one squashed all commits except the one that needed to be reverted. (I forgot to simply revert it immediately.)
- Second one: reverted the commit, and squashed again.

Feedback in https://github.com/unicode-org/icu/pull/1544#issuecomment-857846362 to be bumped to a follow-up PR (whether by me or someone else -- I'm guessing I'll be taking a stab at it.)

**Review by markusicu — APPROVED** [review]
rslgtm

**Pull request merged**
